File: lecture-examples/buffer-overflows/fruits.c
#include <stdio.h>
#include <stdlib.h>

int main() {
    char* buffer[5] = { 
        "apple",
        "orange",
        "pear",
        "banana",
        "tangerine"
    };
    char answer[5];
    int choice;

    printf("Please make a choice:\n");

    for(int i = 0 ; i < 5; ++i) {
        printf("%d. %s\n", i+1, buffer[i]);
    }

    fgets(answer, 5, stdin);
    choice = atoi(answer);

    printf("Here you go: One %s for you!\n",buffer[choice-1]);

    return 0;
}

File: lecture-examples/buffer-overflows/Fruits.java
import java.util.Scanner;

public class Fruits {
    static String[] buffer = {
        "apple",
        "orange",
        "pear",
        "banana",
        "tangerine"
    };

    public static void main(String[] args) {
        System.out.println("Choose something:");

        for(int i = 0; i < buffer.length ; ++i) {
            System.out.println((i + 1) + ". " + buffer[i]);
        }

        int choice = (new Scanner(System.in)).nextInt();

        System.out.println("Here you go: One " + buffer[choice-1] + " for you!");
    }
}
